comp/logs/agent/scheduler_provider.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package agent
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/schedulers"
+	"go.uber.org/fx"
+)
+
+// SchedulerProvider provides a scheduler for the log Agent.
+type SchedulerProvider struct {
+	fx.Out
+
+	SchedulerProvider schedulers.Scheduler `group:"log-agent-scheduler"`
```

Comment:
**💭 thought:** ‏it might be the convetion but why is the varaible named `SchedulerProvider` instead of `Scheduler` ? (it's probably because i'm not used to it but having a `ScheduleProvider` variable within the `SchedulerProvider` struct made me confused)

Reply:
I have renamed `SchedulerProvider` to `Scheduler`.